docs/resources/server/enchantments/index.md

```diff
@@ -0,0 +1,287 @@
+    // `exampleData` is an Increment instance.
+    // This line actually performs the value adjustment.
+    // Each time it runs, mutableValue is set to (exampleData's value) + mutableValue + enchantLevel.
+));
+
+// Use mutableValue elsewhere in your game logic.
+```
+</TabItem>
+
+<TabItem value="explainer" label="Explanation">
+Consider an Enchantment Effect Component called `INCREMENT`, defined as `List<ConditionalEffect<Increment>>`. Let the `Increment` object it contains be a wrapper around an integer that defined a method `add(int x)`, which adds its internal value to the provided integer and returns the result. Imagine that you want to use this object to increase the count of another integer value within your item's `use` method -- say, to increase the number of times it performs some repeated effect.
+
+First, invoke one of the overloads of `EnchantmentHelper#runIterationOnItem`. This function accepts an `EnchantmentHelper.EnchantmentVisitor`, which is a functional interface that accepts an enchantment and its level, and is invoked on all of the enchantments that the given itemstack has. While any consumer of those two values will work, the `Enchantment` class provides a handy function that fits this interface (and is used often by vanilla) -- `Enchantment#applyEffects`. This method is used as above to test the conditions of the `ConditionalEffect`s.
+
+To actually perform the adjustment, use the provided `Increment#add` method.
+
+Note that in this example, the level of the enchantment does not affect the outcome. This can be changed by using `enchantLevel` somewhere in the `Consumer<T>` lambda expression (the last line with code in the example). Any other information stored in the `ItemStack` can also be accessed from here, so other Data Components could be used to inform how the adjustment goes.
+</TabItem>
+</Tabs>
+
+## Enchantment Data Generation
+Enchantment JSON files can be created automatically using the [data generation] system by passing a `RegistrySetBuilder` into `DatapackBuiltInEntriesProvider`. The JSON will be placed in `<project root>/src/generated/data/<modid>/enchantment/<path>.json`.
+
+For more information on how `RegistrySetBuilder` and `DatapackBuiltinEntriesProvider` work, please see the article on [Data Generation for Datapack Registries]. 
+
+<Tabs>
+  <TabItem value="datagen" label="Datagen">
+
+```java
+// This RegistrySetBuilder should be passed into a DatapackBuiltinEntriesProvider in your GatherDataEvent handler.
+public static RegistrySetBuilder BUILDER = new RegistrySetBuilder();
+
+// Define the ResourceKey for our enchantment.
+static ResourceKey<Enchantment> EXAMPLE_ENCHANTMENT_KEY = ResourceKey.create(
+        Registries.ENCHANTMENT,
+        ResourceLocation.fromNamespaceAndPath("examplemod", "example_enchantment")
+);
+
+// Specify the enchantment definition of for our enchantment. 
+static Enchantment.EnchantmentDefinition EXAMPLE_ENCHANTMENT_DEFINITION = new Enchantment.EnchantmentDefinition(
```

Comment:
It should be inlined as the definition can take in a `HolderSet`, so it can reference the bootstraps tag getter. Sure, you can static if you understand what you're doing, but the readers don't, so they should be provided the method in a way to avoid headaches.